pkg/resources/management.cattle.io/v3/setting/validator.go

```diff
@@ -96,20 +99,58 @@ func (a *admitter) Admit(request *admission.Request) (*admissionv1.AdmissionResp
 	return admission.ResponseAllowed(), nil
 }
 
+func (a *admitter) getAuthUserSessionTTLDuration() (time.Duration, error) {
+	setting, err := a.settingCache.Get("auth-user-session-ttl-minutes")
+	if err != nil {
+		return 0, fmt.Errorf("failed to get auth-user-session-ttl-minutes setting: %w", err)
+	}
+	value := effectiveValue(*setting)
+	if value == "" {
+		return 0, nil
+	}
+
+	minutes, err := strconv.ParseInt(value, 10, 64)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse auth-user-session-ttl-minutes setting: %w", err)
+	}
+
+	return time.Duration(minutes) * time.Minute, nil
+}
+
+func (a *admitter) isLessThanUserSessionTTL(dur time.Duration) bool {
+	authUserSessionTTLDuration, getErr := a.getAuthUserSessionTTLDuration()
+	if getErr != nil {
+		logrus.Warnf("failed to get auth-user-session-ttl-minutes setting: %v", getErr) // Log the error and move on.
+	}
+
+	return dur < authUserSessionTTLDuration
```

Comment:
We may potentially want to make sure the `dur` is reasonably larger than the `auth-user-session-ttl-minutes` (e.g. 2x). 🤔

Reply:
Having a dur that is 1 min longer than ttl isn't likely to be useful, but it's not something I feel passionate about when it comes to "we must enforce this".